### DISCUSSION TITLE
jsxInject importSource

### DISCUSSION THREAD

**Discussion by aelbore** [swc-project/swc#2662] [Q&A]
Any suggestion what config should be? Im using atomicojs another jsx library

i want to inject this
```typescript
import {h as _jsx } from 'atomico'
```

my config is like this (what config should i add to have import inject?)
```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "tsx": true
    },
    "transform": {
      "react": {
        "pragma": "_jsx"
      }
    },
    "target": "es2020"
  }
}
```

**Comment by kwonoj** [comment]
`jsc.transform.react.importSource` maybe?

https://swc.rs/docs/configuring-swc#jsctransformreactimportsource

**Reply by aelbore** [reply]
i already tried that it always add runtime for example

atomico/runtime which should not.

**Reply by aelbore** [reply]
maybe alternatively swc plugin will works, ast manipulation

**Reply by kdy1** [reply]
> atomico/runtime which should not.

There's no way to do so.


Note that plugin is very slow.

**Reply by aelbore** [reply]
@kdy1 noted, is this an issue  need to log? there are some library that uses jsx, like preact, solid or atomico.

I added this to `transform.react`
```json
        "runtime": "automatic",
        "importSource": "atomico"
```

output is 
```javascript
import { jsx as _jsx, jsxs as _jsxs } from "atomico/jsx-runtime";
```

expected should be
```javascript
import {h as _jsx } from 'atomico'
```

in esbuild they have this config
```json
   {
 jsxFactory: "_jsx",
    jsxInject: `import {h as _jsx, css as _css} from 'atomico'`,
   }
```
